### ISSUE TITLE
[BUG] Transaction error handler for @Transactional annotation work wrong

### ISSUE THREAD

**Issue by 1000happiness** [spring-projects/spring-data-redis#2359]
The normal redis server allways reply OK for MULTI command. But when I use a proxy with *no* multi command support, the redisTemplate will be caught in error state. 

In specific, after the proxy reply a standard error reply (like "-ERR, unknow command"), the redisTemplate will maintain an error conenction with redis, and dispatch a little request command in this connection. Then the redisTemplate will IGNORE any reply in this error connection.

In conclusion, the redisTemplate do not catch error reply for multi command correctly when using it @Transactional annotation.

**Comment by mp911de** [comment]
You must not use transactional Redis support if you use a server (or middleware) that does not support transactions. Can you provide a reproducer or further details so we can investigate whether we can improve something on our side?

**Comment by spring-projects-issues** [comment]
If you would like us to look at this issue, please provide the requested information. If the information is not provided within the next 7 days this issue will be closed.

**Comment by spring-projects-issues** [comment]
Closing due to lack of requested feedback. If you would like us to look at this issue, please provide the requested information and we will re-open the issue.
